pkg/glance/volumes.go

```diff
@@ -64,60 +51,46 @@ func GetVolumes(name string, pvcName string, secretNames []string, extraVol []gl
 				},
 			},
 		},
+		{
+			Name: "config-data-merged",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{Medium: ""},
+			},
+		},
 	}
 
 	for _, exv := range extraVol {
 		for _, vol := range exv.Propagate(svc) {
 			vm = append(vm, vol.Volumes...)
 		}
 	}
-	secretConfig, _ := GetConfigSecretVolumes(secretNames)
-	vm = append(vm, secretConfig...)
-	return vm
-}
-
-// getInitVolumeMounts - general init task VolumeMounts
-func getInitVolumeMounts(secretNames []string, extraVol []glancev1.GlanceExtraVolMounts, svc []storage.PropagationType) []corev1.VolumeMount {
-	vm := []corev1.VolumeMount{
-		{
-			Name:      "scripts",
-			MountPath: "/usr/local/bin/container-scripts",
-			ReadOnly:  true,
-		},
-		{
-			Name:      "config-data",
-			MountPath: "/var/lib/config-data/default",
-			ReadOnly:  true,
-		},
-		{
-			Name:      "config-data-merged",
-			MountPath: "/var/lib/config-data/merged",
-			ReadOnly:  false,
-		},
-	}
 
-	for _, exv := range extraVol {
-		for _, vol := range exv.Propagate(svc) {
-			vm = append(vm, vol.Mounts...)
+	if len(confSecret) > 0 {
+		//Append the resulting secret
+		for _, secret := range confSecret {
+			sec, _ := GetConfigSecretVolumes(secret)
+			vm = append(vm, sec)
 		}
 	}
-	_, secretConfig := GetConfigSecretVolumes(secretNames)
-	vm = append(vm, secretConfig...)
 	return vm
 }
 
 // GetVolumeMounts - general VolumeMounts
-func GetVolumeMounts(secretNames []string, extraVol []glancev1.GlanceExtraVolMounts, svc []storage.PropagationType) []corev1.VolumeMount {
+func GetVolumeMounts(
+	extraVol []glancev1.GlanceExtraVolMounts,
+	svc []storage.PropagationType,
+	confSecret []string,
+) []corev1.VolumeMount {
 
 	vm := []corev1.VolumeMount{
 		{
-			Name:      "scripts",
-			MountPath: "/usr/local/bin/container-scripts",
-			ReadOnly:  true,
+			Name:      "config-data",
+			MountPath: "/var/lib/config-data/merged",
+			ReadOnly:  false,
```

Comment:
this could be also good time to make volume mounts immutable, or allowing changes in Debug mode only